scripts/update_configs.py

```diff
@@ -86,23 +100,94 @@ def _apply_changes_based_on_config(
+        'are changed else enter n/no.')
+    is_terms_of_service_changed = python_utils.INPUT().lower()
+    while is_terms_of_service_changed not in ['y', 'ye', 'yes', 'n', 'no']:
+        python_utils.PRINT(
+            'Invalid Input: %s. Please enter yes or no.' % (
+                is_terms_of_service_changed))
+        is_terms_of_service_changed = python_utils.INPUT().lower()
+
+    if is_terms_of_service_changed in feconf.AFFIRMATIVE_CONFIRMATIONS:
+        python_utils.PRINT(
+            'Enter sha of the commit which changed the terms of service.')
+        commit_sha = python_utils.INPUT().lstrip().rstrip()
+        commit_time = repo.get_commit(commit_sha).commit.committer.date
+        time_tuple = (
+            commit_time.year, commit_time.month, commit_time.day,
+            commit_time.hour, commit_time.minute, commit_time.second)
+        feconf_lines = []
+        with python_utils.open_file(LOCAL_FECONF_PATH, 'r') as f:
+            feconf_lines = f.readlines()
+        with python_utils.open_file(LOCAL_FECONF_PATH, 'w') as f:
+            for line in feconf_lines:
+                if line.startswith('REGISTRATION_PAGE_LAST_UPDATED_UTC'):
+                    line = (
+                        'REGISTRATION_PAGE_LAST_UPDATED_UTC = '
+                        'datetime.datetime(%s, %s, %s, %s, %s, %s)\n' % (
+                            time_tuple))
+                f.write(line)
+
+
+def add_mailgun_api_key():
+    """Adds mailgun api key to feconf.py."""
+    mailgun_api_key = getpass.getpass(
+        prompt=('Enter mailgun api key from the release process doc.'))
+
+    if re.match('^key-[a-z0-9]{32}$', mailgun_api_key) is None:
+        raise Exception('Invalid mailgun api key.')
+
+    feconf_lines = []
+    with python_utils.open_file(LOCAL_FECONF_PATH, 'r') as f:
+        feconf_lines = f.readlines()
+
+    feconf_lines.append('MAILGUN_API_KEY = \'%s\'\n' % mailgun_api_key)
+    with python_utils.open_file(LOCAL_FECONF_PATH, 'w') as f:
+        for line in feconf_lines:
+            f.write(line)
+
+
+def main():
+    """Updates the LOCAL_FECONF_PATH and 'constants.ts' files after doing the
     prerequisite checks.
     """
     # Do prerequisite checks.
     common.require_cwd_to_be_oppia()
     assert common.get_current_branch_name().startswith('release-')
     common.ensure_release_scripts_folder_exists_and_is_up_to_date()
 
-    _apply_changes_based_on_config(
-        LOCAL_FECONF_PATH, FECONF_CONFIG_PATH, '^([A-Z_]+ = ).*$')
-    _apply_changes_based_on_config(
-        LOCAL_CONSTANTS_PATH, CONSTANTS_CONFIG_PATH, '^(  "[A-Z_]+": ).*$')
+    try:
+        check_updates_to_terms_of_service()
+        add_mailgun_api_key()
+
+        apply_changes_based_on_config(
+            LOCAL_FECONF_PATH, FECONF_CONFIG_PATH, FECONF_REGEX)
+        apply_changes_based_on_config(
+            LOCAL_CONSTANTS_PATH, CONSTANTS_CONFIG_PATH, CONSTANTS_REGEX)
+    except Exception as e:
```

Comment:
There should be a "finally: revert feconf" here (or somewhere) to prevent the mailgun key being persisted, right?

Reply:
I think user needs to deploy after this update, so we cannot revert the changes in this file. The changes will be reverted after the deploying. I will add the tests and methods for that as well in this PR itself.

Reply:
On second thoughts, I feel that is a better idea to add the tests and changes to deploy script in PR #7790 since I am adding tests for deploy script in that PR and making several changes to the structure of deploy script. Is that ok?

Reply:
Yes, that's fine. Perhaps elaborate on the description of PR #7790 a bit to explain what tests are being added.

Reply:
Done.